lib/server/services/user/getUser.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { utilities: { QueryBuilder } } = require('../../../database');
+const UserRepository = require('../../../database/repositories/UserRepository.js');
+
+/**
+ * Find and return an user model by its id
+ *
+ * @param {string} userId the id of the user to find
+ * @param {function|null} qbConfiguration function called with the user find query builder as parameter to add specific configuration to
+ *     the query
+ * @return {Promise<SequelizeUser|null>} the user found or null
+ */
+exports.getUser = (userId, qbConfiguration = null) => {
+    const queryBuilder = new QueryBuilder();
+
+    queryBuilder.where('id').is(userId);
```

Comment:
I believe the `id` fields in the `users` table is the auto-incremented one.
In this scenario, we expect the `external_id` to be received from session DTO and we should query based on that

Reply:
Changed to allow for externalId too